### PR TITLE
[Chronos][IN-372][IN-478][Preprints] Add Chronos Submission

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

This PR adds the ability for preprint creators to submit to Chronos.

## Summary of Changes/Side Effects

Three components, `chronos-submission-status-list`, `chronos-submission-status-list-row` and `chronos-submission-panel` are added.

## Testing Notes

1. Only admins of a preprint can submit to Chronos.
2. Only contributors of the preprint can view all submission statuses. Non-contribs can only see `accepted` or `published` submissions.
3. The preprint has to be accepted by a moderator before it can be submitted to Chronos.

## Ticket

https://openscience.atlassian.net/browse/IN-372

## Notes for Reviewer

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/components/chronos-submission-panel/template.hbs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {{/cp-panel-toggle}}
+
+    {{#cp-panel-body}}
+        <div>
+            {{#power-select search=(perform _fetchInitialJournals) optionsComponent=(component 'lazy-options' loadMore=(action "getMoreJournals") isLoading=isLoading canLoadMore=canLoadMore) options=journals placeholder="Select a journal" selected=selectedJournal onfocus=(action "getInitialJournals" "") oninput=(action "getInitialJournals") onchange=(action "journalSelected") as |journal|}}
+                <strong> {{journal.title}}</strong>
+            {{/power-select}}
+        </div>
+        {{#if selectedJournal}}
+            <div style="padding-top: 5px">
```
Comment (alexschiller) [inline app/components/chronos-submission-panel/template.hbs]: add a class

File: app/components/chronos-submission-panel/template.hbs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                <p>A new tab will open to complete submission on Chronos</p>
+            </div>
+
+            <div class="pull-right">
+                <button class="btn btn-default" onClick={{action 'cancelSubmission'}}>Cancel</button>
+                <button class="btn btn-success" onClick={{perform _submit}}>Submit</button>
+            </div>
+        {{/if}}
+    {{/cp-panel-body}}
+{{/cp-panel}}
```
Comment (alexschiller) [inline app/components/chronos-submission-panel/template.hbs]: end line

File: app/components/chronos-submission-panel/template.hbs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{{#cp-panel}}
+    {{#cp-panel-toggle}}
+        <p>Submit to journals on Chronos</p>
+    {{/cp-panel-toggle}}
+
+    {{#cp-panel-body}}
+        <div>
+            {{#power-select search=(perform _fetchInitialJournals) optionsComponent=(component 'lazy-options' loadMore=(action "getMoreJournals") isLoading=isLoading canLoadMore=canLoadMore) options=journals placeholder="Select a journal" selected=selectedJournal onfocus=(action "getInitialJournals" "") oninput=(action "getInitialJournals") onchange=(action "journalSelected") as |journal|}}
+                <strong> {{journal.title}}</strong>
```
Comment (alexschiller) [inline app/components/chronos-submission-panel/template.hbs]: is the space after strong intended?

File: app/components/chronos-submission-status-list-row/component.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+// import { task } from 'ember-concurrency';
```
Comment (alexschiller) [inline app/components/chronos-submission-status-list-row/component.js]: comment

File: app/components/chronos-submission-status-list-row/template.hbs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div>
+    {{fa-icon statusIcon class=statusIconClass}} {{statusLanguage}}
+    {{#if isContributor}}
+        <a target="_blank" class="fa fa-external-link" href={{submission.submissionUrl}}></a>
+    {{/if}}
+</div>
```
Comment (alexschiller) [inline app/components/chronos-submission-status-list-row/template.hbs]: end line

File: app/components/chronos-submission-status-list/template.hbs [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    {{#each submissions as |submission|}}
+        {{#if isContributor}}
+            {{chronos-submission-status-list-row submission=submission isContributor=isContributor}}
+        {{else}}
+            {{#if (or (eq submission.status "ACCEPTED") (eq submission.status "PUBLISHED"))}}
+                {{chronos-submission-status-list-row submission=submission isContributor=isContributor}}
+            {{/if}}
+        {{/if}}
+    {{/each}}
+{{/if}}
```
Comment (alexschiller) [inline app/components/chronos-submission-status-list/template.hbs]: end line

File: app/components/chronos-submission-panel/component.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        this.set('isLoading', true);
+        const journals = yield this.get('store').query('chronos-journal', { page: 1, 'filter[title]': keyword });
+        this.set('isLoading', false);
+        const canLoadMore = !(journals.get('meta.total_pages') === this.get('page'));
+        // Set the keyword
+        this.set('keyword', keyword);
+        this.set('canLoadMore', canLoadMore);
+        this.set('journals', journals);
+    }).restartable(),
+    actions: {
```
Comment (alexschiller) [inline app/components/chronos-submission-panel/component.js]: do these need analytics? Question for demo

Comment (adlius) [reply app/components/chronos-submission-panel/component.js]: Could go into an improvement ticket.

File: app/components/chronos-submission-panel/template.hbs (hunk 1 above)
Comment (adlius) [reply app/components/chronos-submission-panel/template.hbs]: Adding a new CSS class for the sole purpose of adding a 5px top padding doesn't sound ideal. I've checked elsewhere in the codebase and in-line styling seems to be used for these kind of paddings.

Comment (alexschiller) [reply app/components/chronos-submission-panel/template.hbs]: convincing enough :+1:

**Pull request closed without merging**